src/sage/doctest/parsing.py

```diff
@@ -80,111 +94,455 @@ def fake_RIFtol(*args):
+        ....:         print('|' + line)
+        sage: print_with_ruler([  # the tags are obscured in the source file to avoid relint warnings
+        ....:     update_optional_tags('    sage: something()  # opt' 'ional - latte_int',
+        ....:                          remove_tags=['latte_int', 'wasnt_even_there']),
+        ....:     update_optional_tags('    sage: nothing_to_be_seen_here()',
+        ....:                          tags=['scipy', 'long time']),
+        ....:     update_optional_tags('    sage: nothing_to_be_seen_here(honestly=True)',
+        ....:                          add_tags=['scipy', 'long time']),
+        ....:     update_optional_tags('    sage: nothing_to_be_seen_here(honestly=True, very=True)',
+        ....:                          add_tags=['scipy', 'long time']),
+        ....:     update_optional_tags('    sage: no_there_is_absolutely_nothing_to_be_seen_here_i_am_serious()#opt' 'ional:bliss',
+        ....:                          add_tags=['scipy', 'long time']),
+        ....:     update_optional_tags('    sage: ntbsh()  # abbrv for above#opt' 'ional:bliss',
+        ....:                          add_tags={'scipy': None, 'long time': '30s on the highest setting'}),
+        ....:     update_optional_tags('    sage: no_there_is_absolutely_nothing_to_be_seen_here_i_am_serious()  # really, you can trust me here',
+        ....:                          add_tags=['scipy']),
+        ....: ])
+        |                                                V       V       V       V       V   V   v           v                   v                                       v
+        |    sage: something()
+        |    sage: nothing_to_be_seen_here()             # long time                             # needs scipy
+        |    sage: nothing_to_be_seen_here(honestly=True)        # long time                     # needs scipy
+        |    sage: nothing_to_be_seen_here(honestly=True, very=True)     # long time             # needs scipy
+        |    sage: no_there_is_absolutely_nothing_to_be_seen_here_i_am_serious()         # long time, optional - bliss, needs scipy
+        |    sage: ntbsh()  # abbrv for above            # long time (30s on the highest setting), optional - bliss, needs scipy
+        |    sage: no_there_is_absolutely_nothing_to_be_seen_here_i_am_serious()  # really, you can trust me here                # needs scipy
+
+    When no tags are changed, by default, the unchanged input is returned.
+    We can force a rewrite; unconditionally or whenever standard tags are involved.
+    But even when forced, if comments are already aligned at one of the standard alignment columns,
+    this alignment is kept even if we would normally realign farther to the left::
+
+        sage: print_with_ruler([
+        ....:     update_optional_tags('    sage: unforced()       # opt' 'ional - latte_int'),
+        ....:     update_optional_tags('    sage: unforced()  # opt' 'ional - latte_int',
```

Comment:
Splitting the string containing `# optional` is still necessary? The doctester or the fixer is not robust enough yet to handle the case?

Reply:
It is `tox -e relint` that is not smart enough

Reply:
I see. OK.